lib/matplotlib/rcsetup.py

```diff
@@ -1286,6 +1286,19 @@ def _validate_linestyle(ls):
     'figure.subplot.hspace': [0.2, ValidateInterval(0, 1, closedmin=True,
                                                      closedmax=False)],
 
+    # do constrained_layout.
+    'figure.constrainedlayout.do': [False, validate_bool],
```

Comment:
Somehow the "do" grates on my aesthetic senses.  Maybe "use" or "enable" or "active"?  Regardless, the matplotlibrc.template also needs to be updated, since it presently has only "figure.constrainedlayout".
Also, since the kwarg is "constrained_layout", and that form (with underscore) seems prevalent (and readable), maybe the rcParam family name should match it.  Unless we can think of something shorter... It's too bad "autolayout" is already used.